test/TestEnvironment.js

```diff
@@ -0,0 +1,31 @@
+const NodeEnvironment = require('jest-environment-node');
+const { unlink, writeFile } = require('fs');
```

Comment:
Why aren't we using mocked fs?

Reply:
Because some tests need to get the actual files for them to work. For instance the one that tests for how many rule files are there.

Reply:
For those you can use `jest.requireActual`, right?

Reply:
there is no way for the custom jest env to distinguish between tests inside the same file.
I would like to keep it as it is.

Reply:
Sure, there's no way to distinguish it here and it shouldn't be done here, rather in test files. I don't really understand how can actual fs writes work when tests run in parallel, doesn't it cause race conditions? FS operations are also quite slow.

Reply:
we are using pragma marks to only write that file for that test.